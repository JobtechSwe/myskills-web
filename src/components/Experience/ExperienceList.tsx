import React from 'react'
import styled from '@emotion/styled'

const List = styled.ul`
  color: white;
  font-family: 'Arial';
  padding: 0;
  font-weight: bold;
  text-align: center;
  li {
    margin-bottom: 5px;
    list-style: none;
    transition: all 100ms ease;

    &:hover {
      transform: scale(1.1);
    }
  }
`

const ExperienceList = ({ experiences, handleSetExperiences }: any) => {
  return (
    <List>
      {experiences
        .filter(({ taxonomyId }: { taxonomyId: string }) =>
          experiences.some(
            (y: { taxonomyId: string }) => y.taxonomyId === taxonomyId
          )
        )
        .map((c: { term: string }, i: number) => (
          <li key={i}>
            <button onClick={() => handleSetExperiences(c)}>{c.term}</button>
          </li>
        ))}
    </List>
  )
}

export default ExperienceList
