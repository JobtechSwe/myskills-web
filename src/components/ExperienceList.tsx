import React from 'react'
import { TaxonomyDefaultResult } from '../generated/myskills'
import List from './List'

type MutationVariables = {
  variables: {
    experience: {
      term: string
      years: string
      taxonomyId: string
    }
  }
}

interface ExperienceListProps {
  list: TaxonomyDefaultResult[]
  addExperience?: (variables: MutationVariables) => void
}

const ExperienceList: React.FC<ExperienceListProps> = ({
  list,
  addExperience,
}) => {
  return (
    <List>
      {list.map((experience: TaxonomyDefaultResult, i: number) => {
        return (
          <li key={i}>
            {addExperience ? (
              <button
                onClick={() =>
                  addExperience({
                    variables: {
                      experience: {
                        term: experience.term,
                        years: '',
                        taxonomyId: experience.taxonomyId,
                      },
                    },
                  })
                }
              >
                {experience.term}
              </button>
            ) : (
              <p>{experience.term}</p>
            )}
          </li>
        )
      })}
    </List>
  )
}

export default ExperienceList
