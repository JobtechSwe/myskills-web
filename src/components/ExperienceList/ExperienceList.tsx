import React from 'react'
import styled from '@emotion/styled'
import { Taxonomy } from '../../generated/myskills'

type MutaionVariables = {
  variables: {
    experience: {
      name: string
      years: string
      taxonomyId: string
    }
  }
}

const List = styled.ul`
  color: white;
  font-family: 'Arial';
  padding: 0;
  font-weight: bold;
  text-align: center;
  li {
    margin-bottom: 5px;
    list-style: none;
    transition: all 100ms ease;

    &:hover {
      transform: scale(1.1);
    }
  }
`

interface ExperienceListProps {
  list: Taxonomy.Result[]
  addExperience?: (variables: MutaionVariables) => void
}

const ExperienceList: React.FC<ExperienceListProps> = ({
  list,
  addExperience,
}) => {
  return (
    <List>
      {list.map((experience: Taxonomy.Result, i: number) => {
        return (
          <li key={i}>
            {addExperience ? (
              <button
                onClick={() =>
                  addExperience({
                    variables: {
                      experience: {
                        name: experience.term,
                        years: '',
                        taxonomyId: experience.taxonomyId,
                      },
                    },
                  })
                }
              >
                {experience.term}
              </button>
            ) : (
              <p>{experience.term}</p>
            )}
          </li>
        )
      })}
    </List>
  )
}

export default ExperienceList
