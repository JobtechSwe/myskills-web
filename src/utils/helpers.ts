import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
