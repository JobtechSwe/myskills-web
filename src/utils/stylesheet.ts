const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
