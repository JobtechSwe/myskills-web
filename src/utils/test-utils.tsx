import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
