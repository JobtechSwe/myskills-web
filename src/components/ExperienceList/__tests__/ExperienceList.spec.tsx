import * as React from 'react'
import ExperienceList from '../ExperienceList'
import { render } from '../../../utils/test-utils'
import { wait, fireEvent } from 'react-testing-library'

describe('views/ExperienceList', () => {
  let experiences: any
  let addExperience: any

  beforeEach(() => {
    addExperience = jest.fn()
    experiences = [
      {
        term: 'Javascript',
        taxonomyId: 'abc',
      },
      {
        term: 'Java',
        taxonomyId: 'cda',
      },
    ]
  })

  it('renders empty list', async () => {
    const { container } = render(<ExperienceList list={[]} />)

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('renders experineces list', async () => {
    const { container } = render(<ExperienceList list={experiences} />)

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('renders experineces list with button', async () => {
    const { container } = render(
      <ExperienceList addExperience={addExperience} list={experiences} />
    )

    await wait()

    expect(container).toMatchSnapshot()
  })

  it('should render with taxonomy query result', async () => {
    const { getByText } = render(
      <ExperienceList addExperience={addExperience} list={experiences} />
    )

    fireEvent.click(getByText(/java/i))

    await wait()

    expect(addExperience).toHaveBeenCalledWith({
      variables: {
        experience: { term: 'Javascript', taxonomyId: 'abc', years: '' },
      },
    })
  })
})
