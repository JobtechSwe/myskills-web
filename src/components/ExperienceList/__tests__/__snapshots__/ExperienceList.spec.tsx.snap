// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/ExperienceList renders empty list 1`] = `
<div>
  <ul
    class="css-1n9h91p"
    color="persianBlue"
  />
</div>
`;

exports[`views/ExperienceList renders experineces list 1`] = `
<div>
  <ul
    class="css-1n9h91p"
    color="persianBlue"
  >
    <li>
      <p>
        Javascript
      </p>
    </li>
    <li>
      <p>
        Java
      </p>
    </li>
  </ul>
</div>
`;

exports[`views/ExperienceList renders experineces list with button 1`] = `
<div>
  <ul
    class="css-1n9h91p"
    color="persianBlue"
  >
    <li>
      <button>
        Javascript
      </button>
    </li>
    <li>
      <button>
        Java
      </button>
    </li>
  </ul>
</div>
`;
